packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
+      widget.controller?._attach(this);
     });
   }
 
   @override
   void dispose() {
     super.dispose();
-    _animationController.dispose();
+    widget.controller?._detach();
+    _swipeActivity?.animation.dispose();
   }
 
   @override
-  Widget build(BuildContext context) {
-    return LayoutBuilder(
-      builder: (BuildContext context, BoxConstraints constraints) {
-        return Container(
-          padding: widget.padding,
-          child: LayoutBuilder(
-            builder: (BuildContext context, BoxConstraints constraints) {
-              return Stack(
-                  clipBehavior: Clip.none,
-                  fit: StackFit.expand,
-                  children: [
-                    if (widget.loop || currentIndex < widget.cardsCount - 1)
-                      ..._backgroundCards(constraints),
-                    if (currentIndex < widget.cardsCount)
-                      _foregroundItem(constraints)
-                  ]);
-            },
-          ),
-        );
-      },
-    );
+  void didUpdateWidget(covariant AppinioSwiper oldWidget) {
+    if (oldWidget.cardCount != widget.cardCount ||
+        oldWidget.threshold != widget.threshold ||
+        oldWidget.maxAngle != widget.maxAngle ||
+        oldWidget.invertAngleOnBottomDrag != widget.invertAngleOnBottomDrag ||
+        oldWidget.loop != widget.loop) {
+      position._updateFromWidgetState(widget);
+      widget.controller?._attach(this);
+    }
+    if (oldWidget.controller != widget.controller) {
+      oldWidget.controller?._detach();
+      widget.controller?._attach(this);
+    }
+    super.didUpdateWidget(oldWidget);
   }
 
-  List<Widget> _backgroundCards(BoxConstraints constraints) {
-    List<Widget> backgroundCards = [];
-    int i = currentIndex + 1;
-    int j = 1;
-    double difference = _difference;
-    double scale = _scale;
-    while ((i < widget.cardsCount || widget.loop) &&
-        j <= widget.backgroundCardsCount) {
-      backgroundCards.add(_backgroundItem(constraints, i, difference, scale));
-      difference += _backgroundCardsDifference;
-      scale -= _backgroundCardsScaleDifference;
-      i++;
-      j++;
-    }
-    return backgroundCards.reversed.toList();
-  }
-
-  Widget _backgroundItem(
-      BoxConstraints constraints, int index, double difference, double scale) {
-    return Positioned(
-      top: difference,
-      left: 0,
-      child: Container(
-        color: Colors.transparent,
-        child: Transform.scale(
-          scale: scale,
-          child: Container(
-            constraints: constraints,
-            child: widget.cardsBuilder(
-              context,
-              (index % widget.cardsCount),
-            ),
-          ),
-        ),
-      ),
-    );
+  @override
+  void didChangeDependencies() {
+    position._cardSize = MediaQuery.of(context).size;
+    super.didChangeDependencies();
   }
 
-  Widget _foregroundItem(BoxConstraints constraints) {
-    return Positioned(
```

Comment:
`Transform.translate` is equivalent and slightly cleaner than `Positioned` and passing the parent constraints through.